src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -563,6 +578,49 @@ private static int compareContainerOrder(String importName1, String importName2,
     }
 
     /**
+     * Compares two import strings.
+     * We first compare the container of the import, container being the type enclosing
+     * the element being imported. When this returns 0, we compare the qualified
+     * import name considering a type-import-on-demand declaration last in group.
+     * @param importName1 first import name.
+     * @param importName2 second import name.
+     * @param caseSensitive whether the comparison of fully qualified import names is case
+     *                      sensitive.
+     * @return the value {@code 0} if str1 is equal to str2; a value
+     *         less than {@code 0} if str is less than the str2 ;
+     *         and a value greater than {@code 0} if str1 is greater than str2
+     */
+    private static int compareOnDemandImport(String importName1, String importName2,
+                                             boolean caseSensitive) {
+        final String container1 = getImportContainer(importName1);
+        final String container2 = getImportContainer(importName2);
```

Comment:
Are you sure that all code from compareContainerOrder is required ? 
You suppose to have only pre-decision before simple compare() execution.